src/library/scala/concurrent/duration/Duration.scala

```diff
@@ -124,6 +124,8 @@ object Duration {
       Undefined
     else if (nanos > Long.MaxValue || nanos < Long.MinValue)
       throw new IllegalArgumentException("trying to construct too large duration with " + nanos + "ns")
+    else if (nanos < 0)
+      fromNanos((nanos - 0.5).toLong)
```

Comment:
I have to add that apparently `java.lang.Math.round(-0.5) == 0` and `java.lang.Math.round(-0.51) == -1`, while in this pull request `Duration(-0.5, NANOSECONDS) == Duration(-1, NANOSECONDS)`. So if this is meant to emulate standard java.lang rounding, just using `java.lang.Math.round` or `scala.math.round` might be better.